scripts/zmq_service.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python3
+
+import psij
+
+import functools
+
+import radical.utils as ru
+
+from typing import List, Dict, Any, Tuple
+
+
+'''
+This file implements a psij job service.  The service can be contacted via ZMQ.
+It listens for requests on a `zmq.REP` reply socket - a client should
+accordingly connect via an `zmq:REQ` request socket.
+
+the service will
+
+Requests are formed as follows:
+
+    {
+        'cmd'   : str,
+        'args'  : Tuple[Any],
+        'kwargs': Dict[str, Any]
+    }
+
+Repsonse messages are formed as follows:
+
+    {
+        'err'   : str = None,  # error message
+        'exc'   : str = None,  # exception stacktrace
+        'res'   : Any          # any serializable data type
```

Comment:
No. Cucumbers are pickleable.

Sorry.

Reply:
Good point - we actually use [`msgpack`](https://msgpack.org/) for serialization as we find pickle to be [slow and insecure](https://www.benfrederickson.com/dont-pickle-your-data/), and also less portable (it would be difficult to write non-Python clients to the service).  I added a statement to that effect.